### PR TITLE
feat(kafka-logger): support sasl config in brokers

### LINKED ISSUE #8046: feat: As a user, I want to support sasl_config in kafka-logger plugin, so that we can send logs to brokers that enable authentication
### Description

Hello,

Now I found "lua-resty-kafka" lib had supported set `sasl_config` in the client.
docs: https://github.com/doujiang24/lua-resty-kafka#new

So I want to support set `sasl_config` in kafka-logger plugin, then we can send logs to brokers that enable authentication.
Like: 

```
{
    "kafka-logger":{
        "brokers":[
            {
                "host":"127.0.0.1",
                "port":9092,
                "sasl_config":{
                    "mechanism":"PLAIN",
                    "user":"USERNAME",
                    "password":"PASSWORD"
                }
            }
        ],
        "kafka_topic":"test2",
        "key":"key1",
        "batch_max_size":1,
        "name":"kafka logger"
    }
}
```

### PR DESCRIPTION
**Pull request by starsz**
### Description

Fixes #8046 

Thanks to @biubiue and the original PR https://github.com/apache/apisix/pull/7782.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by starsz** [comment]
> I think it is not good to put `sasl_config` under `borkers` node, if more than one borker node, you need to configure duplicate `sasl_config`

Hi, @biubiue .I think we can put the `sasl_config` under `brokers` first so that we can support configuring different `sasl_config` under different brokers.

Then we can add `sasl_config` besides `sasl_config` as global `sasl_config`, if the broker doesn't configure `sasl_config` then it will use global sasl_config instead.

File: docs/en/latest/plugins/kafka-logger.md [hunk 1]
```diff
@@ -41,6 +41,10 @@ It might take some time to receive the log data. It will be automatically sent a
...
+| brokers.sasl_config.mechanism  | string  | False    | "PLAIN"          | ["PLAIN"]           |     The mechaism of sasl config                                                                                                                                                                                                                                                                                                             |
+| brokers.sasl_config.user       | string  | True     |                  |                     |    The user of sasl_config.If sasl_config exists, it's required.                                                                                                                                                                                                                                                                                             |
```
Comment (spacewander) [inline docs/en/latest/plugins/kafka-logger.md:46]:
```suggestion
| brokers.sasl_config.user       | string  | True     |                  |                     |  The user of sasl_config. If sasl_config exists, it's required.                                                                                                                                                                                                                                                                                             |
```

File: docs/en/latest/plugins/kafka-logger.md [hunk 2]
```diff
@@ -41,6 +41,10 @@ It might take some time to receive the log data. It will be automatically sent a
...
+| brokers.sasl_config.password   | string  | True     |                  |                     | The password of sasl_config.If sasl_config exists, it's required.                                                                                                                                                                                                                                                                                                 |
```
Comment (spacewander) [inline docs/en/latest/plugins/kafka-logger.md:47]:
```suggestion
| brokers.sasl_config.password   | string  | True     |                  |                     | The password of sasl_config. If sasl_config exists, it's required.                                                                                                                                                                                                                                                                                                 |
```

File: docs/zh/latest/plugins/kafka-logger.md [hunk 3]
```diff
@@ -39,6 +39,10 @@ description: API 网关 Apache APISIX 的 kafka-logger 插件用于将日志作
 | brokers                | array   | 是     |                |                       | 需要推送的 Kafka 的 broker 列表。                   |
 | brokers.host           | string  | 是     |                |                       | Kafka broker 的节点 host 配置，例如 `192.168.1.1`                     |
 | brokers.port           | string  | 是     |                |                       | Kafka broker 的节点端口配置                         |
+| brokers.sasl_config    | object  | 否     |                |                       | Kafka broker 中的 sasl_config                     |
+| brokers.sasl_config.mechanism  | string  | 否     | "PLAIN"          | ["PLAIN"]   | Kafka broker 中的 sasl 认证机制                     |
+| brokers.sasl_config.user       | string  | 是     |                  |             | Kafka broker 中 sasl 配置中的 user， 如果 sasl_config 存在，则必须填写                 |
```
Comment (spacewander) [inline docs/zh/latest/plugins/kafka-logger.md:44]:
```suggestion
| brokers.sasl_config.user       | string  | 是     |                  |             | Kafka broker 中 sasl 配置中的 user，如果 sasl_config 存在，则必须填写                 |
```

File: docs/zh/latest/plugins/kafka-logger.md [hunk 4]
```diff
@@ -39,6 +39,10 @@ description: API 网关 Apache APISIX 的 kafka-logger 插件用于将日志作
...
+| brokers.sasl_config.password   | string  | 是     |                  |             | Kafka broker 中 sasl 配置中的 password， 如果 sasl_config 存在，则必须填写             |
```
Comment (spacewander) [inline docs/zh/latest/plugins/kafka-logger.md:45]:
```suggestion
| brokers.sasl_config.password   | string  | 是     |                  |             | Kafka broker 中 sasl 配置中的 password，如果 sasl_config 存在，则必须填写             |
```

File: apisix/plugins/kafka-logger.lua [hunk 5]
```diff
@@ -68,6 +68,19 @@ local schema = {
                         maximum = 65535,
                         description = "the port of kafka broker",
                     },
+                    sasl_config = {
+                        type = "object",
+                        description = "sasl config",
+                        properties = {
+                            mechanism = {
+                                type = "string",
+                                default = "PLAIN"
```
Comment (spacewander) [inline apisix/plugins/kafka-logger.lua:77]: We can use enum to make sure the passing mechanism is legal? Although only `PLAIN` is currently supported.

**Comment by tzssangglass** [comment]
LGTM

File: apisix/plugins/kafka-logger.lua (hunk 5 above)
Comment (starsz) [reply apisix/plugins/kafka-logger.lua:77]: Good suggestion. Let me add the enum and add a test case to cover it.

**Review by tzssangglass — APPROVED** [review]

File: ci/pod/docker-compose.plugin.yml [hunk 6]
```diff
@@ -104,17 +104,22 @@ services:
...
+      - "19094:9094"
     depends_on:
       - zookeeper-server1
       - zookeeper-server2
     networks:
       kafka_net:
+    volumes:
+      - ./ci/pod/kafka/kafka-server/kafka_jaas.conf:/opt/bitnami/kafka/config/kafka_jaas.conf:ro
+      - ./ci/pod/kafka/kafka-server/selfsigned.jks:/opt/bitnami/kafka/config/certs/kafka.keystore.jks:ro
+      - ./ci/pod/kafka/kafka-server/selfsigned.jks:/opt/bitnami/kafka/config/certs/kafka.truststore.jks:ro
```
Comment (soulbird) [inline ci/pod/docker-compose.plugin.yml:122]: The `selfsigned.jks` file is only generated when the `last` test is executed, look: https://github.com/apache/apisix/blob/master/.github/workflows/build.yml#L117-L120, and should not exist when the `plugin` test is executed. Why can CI still pass, very puzzled.....

Comment (starsz) [reply ci/pod/docker-compose.plugin.yml:122]: Sorry.My fault.

> Why can CI still pass, very puzzled....

The file or directory does not need to exist on the Docker host already. It is created on demand if it does not yet exist.
So it wouldn't influence the CI.

refer: https://docs.docker.com/storage/bind-mounts/

Comment (soulbird) [reply ci/pod/docker-compose.plugin.yml]: Got it ~~~

**Review by soulbird — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**